File: ABM/src_python/collect_data.py
from mesa import Agent

def susceptible(model):
    return sum(1 for a in model.schedule.agents if a.status == 'S')

def exposed(model):
    return sum(1 for a in model.schedule.agents if a.status == 'E')

def infected(model):
    return sum(1 for a in model.schedule.agents if a.status == 'I')

def recovered(model):
    return sum(1 for a in model.schedule.agents if a.status == 'R')

File: ABM/src/model/collect_data.jl
using Agents

total_infected(m) = count(a.status == :I for a ∈ allagents(m))
susceptible(m) = count(a == :S for a ∈ m)
exposed(m) = count(a == :E for a ∈ m)
infected(m) = count(a == :I for a ∈ m)
recovered(m) = count(a == :R for a ∈ m)